src/core/inlines.js

```diff
@@ -118,6 +133,7 @@ export function run(conf) {
       "\\b(?:NOT\\s+)?RECOMMENDED\\b",
       "\\bOPTIONAL\\b",
       "(?:{{3}\\s*.*\\s*}{3})", // inline IDL references,
+      "\\|\\w+(?:\\:\\w+)?\\|", // inline variable regex
```

Comment:
@saschanaz we plan to match on Interface names and primitives (e.g., string, object, promise) etc. Does this look ok?

Reply:
I'm not very familiar with this kind of things... Especially haven't seen scoped variables. Do you have some existing examples?

Reply:
We are inventing something new :) anyway, I think I should trial it in one or two specs. BikeShed uses unscoped ones. Scoped ones offer all sorts of neat opportunities.

Reply:
I want to understand what would it even mean, will there be same-named variables with different scopes? How would it be represented visually?

Reply:
Visually, I’m imagining something similar (but simpler) to what VSCode shows when you hover over a variable. Once the type is declared, all other matching vars inherent the type. This really helpful when reading algorithms, as you can imagine.